src/lib/utils/useDidRegainFocus.tsx

```diff
@@ -0,0 +1,27 @@
+import { useIsFocused } from "@react-navigation/native"
+import { useEffect, useRef } from "react"
+import usePrevious from "react-use/esm/usePrevious"
+
+export function useDidRegainFocus() {
+  const isVisible = useIsFocused()
+  const lastIsVisible = usePrevious(isVisible) ?? isVisible
+  const hasBeenVisibleAtLeastOnce = useRef(false)
+  const didJustBecomeVisible = isVisible && !lastIsVisible
+  const didRegainFocus = didJustBecomeVisible && hasBeenVisibleAtLeastOnce.current
+
+  if (isVisible) {
+    hasBeenVisibleAtLeastOnce.current = true
+  }
+
```

Comment:
this could also be simplified to
```suggestion
  const isVisible = useIsFocused()
  const lastIsVisible = usePrevious(isVisible) ?? true
  const didRegainFocus = isVisible && !lastIsVisible
```

but it's a little less obvious how it works like this.